finetuner/tuner/callback/training_checkpoint.py

```diff
@@ -0,0 +1,143 @@
+import os
+import pickle
+from typing import TYPE_CHECKING
+import shutil
+
+import keras
+import paddle
+import torch
+from jina.logging.logger import JinaLogger
+
+from finetuner.helper import get_framework
+
+from .base import BaseCallback
+
+if TYPE_CHECKING:
+    from ..base import BaseTuner
+
+
+class TrainingCheckpoint(BaseCallback):
+    """
+    Callback to save model at every epoch or the last K epochs
+    """
+
+    def __init__(self, save_dir: str, last_k_epochs: int = 1):
+        """
+        :param save_dir: string, path to save the model file.
+        :param last_k_epochs: this parameter is an integer. Only the most
+            recent k checkpoints will be kept. Older checkpoints are deleted
+        """
+        self._logger = JinaLogger(self.__class__.__name__)
+        self._save_dir = save_dir
+        self._last_k_epochs = last_k_epochs
+        self._saved_checkpoints = []
+        if not save_dir:
+            raise ValueError(
+                '``save_dir`` parameter is mandatory. Pass it in parameters'
+            )
```

Comment:
I think you can delete this - you already made `save_dir` a required parameter by not providing a default